### ISSUE TITLE
Self-signed certs rejected by smime_keys when coupled with a private key

### ISSUE THREAD

**Issue by libBletchley** [neomutt/neomutt#935]
When running: `smime_keys add_p12 my_self-signed_key.p12`, the result is:
```
No root and no intermediate certificates. Can't continue. at /usr/bin/smime_keys line 1080.
```
There are a few bugs here:
1. If a cert is self-signed, `smime_keys` should not expect to find a root or intermediate cert.
2. The `add_p12` command is to add one's **own** key pair.  The private key is in the same container.  Having the private key for a corresponding certificate is as authentic as ownership gets.  Of course a person trusts themself.
3. Even if a user is using a CA-signed key, the duty of authentication is misplaced.  That is, it's the _other party_ who must decide whether a cert is valid based on which CAs _they_ trust, and _their_ willingness to accept a self-signed cert.  What if I only trusted Verisign exclusively, and yet my one cert were signed by Comodo?  It's not me who must trust Comodo, it's my correspondants.

Here is a reproducer:
```
#!/bin/bash -xe

typeset -r cn_email_address=alice@aclu.org
typeset -r cn_name='Alice Walker'
typeset -r keystore_pw='catsname'
typeset -r key_length=${key_length:-4096}
typeset -r dir_keystore=$(mktemp -d)
typeset -r fn_keystore=$dir_keystore/self.p12

pushd "$dir_keystore"

openssl genrsa -passout pass:"$keystore_pw" -des3 -out cert.key "$key_length"
openssl req -passin pass:"$keystore_pw" -new -key cert.key -subj /CN=alice/emailAddress="$cn_email_address" -out cert.csr
openssl x509 -passin pass:"$keystore_pw" -req -days 3650 -in cert.csr -signkey cert.key -out cert.crt
openssl pkcs12 -export -in cert.crt -inkey cert.key -name "$cn_name" -out "$fn_keystore" -passin pass:"$keystore_pw" -passout pass:"$keystore_pw"

printf %s "set smime_ca_location=\"/etc/ssl/certs/\"
set smime_keys=\"$dir_keystore/keys/\"
set smime_certificates=\"$dir_keystore/certificates/\"
" > "$dir_keystore"/muttrc

ln -s "$dir_keystore"/muttrc ~/.muttrc

smime_keys init
smime_keys add_p12 "$fn_keystore"
popd
```
Similarly, importing self-signed certificates supplied by correspondants using `^k` also fails:
```
You may assign a label to this key, so you don't have to remember
the key ID. This has to be _one_ word (no whitespaces).

Enter label: Found 0 certificate chains
Press any key to continue...
```
There are 2 bugs here:
1. Mutt asks the user for a label, and then rejects the certificate.  Why ask for a label if the cert has a problem of any kind?  Users should only be labelling certs that are accepted into the db.
2. It's looking for a certificate chain on a self-signed certificate.  Mutt should not expect a self-signed certificate to have a certificate chain.

**Comment by guygma** [comment]
What is the reason for this being closed? Is it deemed fixed? It does not appear to be from my end, but perhaps the maintainers do not view this as their bug. I have similar issues with neomutt complaining of self-signed certificates even when the root ECA and IdenTrust certificates from the chain are included with `smime_keys add_root`. Its not a problem with the certificates themselves as they work in other email programs.

**Comment by libBletchley** [comment]
@guygma 

Good question.  I think a lot of my bug reports were closed on Aug. 8 without fixing.  I suspect the bugs got stale and developers were swamped and trying to reduce the clutter of bugs found to have low priority.

Would be useful if they were tagged in some way.. like "wontfix", so ppl know the bugs still exist.  Perhaps it should also be documented in the `smime_keys` man page in a "Limitations" section.

**Comment by guygma** [comment]
Got it. I opened a fresh issue that is more descriptive of  the specific problem I am experiencing with the S/MIME functionality. (#1488)

**Comment by flatcap** [comment]
> I suspect the bugs got stale and developers were swamped and trying to reduce the clutter of bugs found to have low priority.

Thanks.  That's very understanding of you.
You're right and I should have been clearer about my actions.

The crypto code is a large part of (neo)mutt that we've left alone.
We don't understand it deeply and 'if it's good enough for mutt...'
That's a bit lazy, I know, but there are far more urgent things to fix.

- There are currently four developers
- I'm a coder, not a manager

I do my best to keep the development focussed, but I find that having hundreds of issues becomes overwhelming.
**I** haven't read all 117 issues in ages (and it's my project).  I doubt anyone else has, either.

Every so often I have a clear out.

Bugs are generally left open and enhancement requests are closed (tagged 'hard').
That leaves the issues that nobody understands, or that aren't valid -- a source of stress.
They tend to get closed with the hope that nobody notices.  Sorry.

**Comment by guygma** [comment]
@flatcap: All very understandable. Thanks for providing context; it helps to understand that. I didn't realize there were really just 4 developers on Neomutt either... thanks for keeping this project healthy!

As for my use of (neo)mutt at work, it turns out enhancing the robustness of S/MIME handling is crucial so I have got some skin in the game, as it were. I run a local IMAP server for all my mails so I that I am not bound to a particular MUA, but I would prefer to have my default be (neo)mutt rather than Thunderbird. So, as I mentioned in #1488, I am quite willing to help on this but since I have no experience with perl some guidance from a dev who does would be appreciate. I think I understand the root cause leading to my issue (and I outlined it in #1488) so at least some progress should be achievable.
